### PR TITLE
fix(40610): "Convert to template string" result unnecessarily complicated if one part was already a template string

### LINKED ISSUE #40610: "Convert to template string" result unnecessarily complicated if one part was already a template string
**TypeScript Version:**  4.1.0-dev.20200909

**Search Terms:** convert template string

**Code**

Obviously, this is a very dumbed-down example, but the same happens for more complicated concatenations.
```ts
const test = "" + ``;
//            ^ put cursor here, trigger "convert to template string"
```

**Expected behavior:**
```ts
const test = ``;
```

**Actual behavior:**
```ts
const test = `${``}`;
```

**Playground Link:** no quick fixes available on the playground

**Related Issues:** none found

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #40610

### CODE REVIEW

**Comment by ExE-Boss** [comment]
What about:
```js
`a${foo} ${bar}` + " other " + baz + ` ${biz}…`
```

---

Which should become:
```js
`a${foo} ${bar} other ${baz} ${biz}…`
```

File: tests/cases/fourslash/refactorConvertStringOrTemplateLiteral_TemplateString2.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/// <reference path='fourslash.ts' />
+
+////const foo = 1;
+////const bar = /*start*/"a" + `${foo}`/*end*/;
```
Comment (jessetrinity) [inline tests/cases/fourslash/refactorConvertStringOrTemplateLiteral_TemplateString2.ts:4]: It looks like the refactor will delete strings following the substitution part, e.g.
```ts
"a" + `${foo}` + "b"
```

**Review by jessetrinity — CHANGES_REQUESTED** [review]

**Comment by sandersn** [comment]
@gabritto you might also be interested in this one.

**Comment by a-tarasyuk** [comment]
@jessetrinity Are there any additional cases that need to cover?

**Review by jessetrinity — APPROVED** [review]
Looks good, thanks!

**Pull request merged**